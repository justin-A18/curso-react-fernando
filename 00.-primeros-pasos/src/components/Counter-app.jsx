import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ({ value }) => {
	const [counter, setCounter] = useState(value);

	function handleClick() {
		setCounter(counter + 1);
	}

	function handleDecremento() {
		setCounter(counter - 1);
	}

	function handleReset() {
		setCounter(value);
	}

	return (
		<>
			<h1>CounterApp</h1>
			<h2> {counter} </h2>
			<button onClick={handleDecremento}>decremento</button>
			<button onClick={handleReset}>resetear</button>
			<button onClick={handleClick}>incremento</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
};
