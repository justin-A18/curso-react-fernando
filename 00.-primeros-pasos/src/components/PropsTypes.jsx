import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle }) => {
	if (!title) {
		throw new Error('title no existe');
	}

	return (
		<>
			<h1 data-testid="test-title">{title}</h1>
			<p>{subtitle}</p>
		</>
	);
};

//TIPADO
FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subtitle: PropTypes.string.isRequired
}

//VALORES POR DEFECTO
FirstApp.defaultProps = {
	//title: "No hay title",
	subtitle: "No hay subtitle"
}