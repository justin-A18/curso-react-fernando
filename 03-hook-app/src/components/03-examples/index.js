export * from './Advice'
export * from './LoadingQuote'