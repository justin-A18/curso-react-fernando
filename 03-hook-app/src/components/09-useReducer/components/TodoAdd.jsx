import PropTypes from 'prop-types';
import { useState } from 'react';

export const TodoAdd = ({ onNewTodo }) => {
	const [inputValue, setInputValue] = useState('');

	const handleChangeInput = ({ target }) => {
		const { value } = target;
		setInputValue(value);
	};

	const handleSubmitForm = (e) => {
		e.preventDefault();

		if (inputValue === '') return;

		const newTodo = {
			id: new Date().getTime(),
			description: inputValue,
			done: false,
		};

		onNewTodo(newTodo);
		setInputValue('');
	};

	return (
		<form
			className='gap-2 flex-column d-flex'
			onSubmit={handleSubmitForm}>
			<input
				type='text'
				placeholder='Â¿Que haremos hoy?'
				className='form-control'
				value={inputValue}
				onChange={handleChangeInput}
			/>

			<button
				type='submit'
				className='btn btn-primary'>
				agregar
			</button>
		</form>
	);
};

TodoAdd.propTypes = {
	onNewTodo: PropTypes.func.isRequired,
};
