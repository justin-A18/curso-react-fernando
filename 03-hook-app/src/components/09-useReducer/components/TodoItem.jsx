import PropTypes from 'prop-types';

export const TodoItem = ({ todo = {}, onDeleteTodo, onToggleTodo }) => {
	const isDone = todo.done ? 'done' : '';
	const isDoneTodo = todo.done ? 'text-decoration-line-through' : '';

	return (
		<li
			key={todo.id}
			data-id={todo.id}
			className={`list-group-item d-flex align-items-center justify-content-between ${isDone}`}>
			<span
				className={`fs-5 ${isDoneTodo}`}
				onClick={() => onToggleTodo(todo.id)}
				aria-label='span'>
				{todo.description}
			</span>

			<button
				className='btn btn-danger'
				onClick={() => onDeleteTodo(todo.id)}>
				Borrar
			</button>
		</li>
	);
};

TodoItem.propTypes = {
	todo: PropTypes.object.isRequired,
	onDeleteTodo: PropTypes.func.isRequired,
	onToggleTodo: PropTypes.func.isRequired,
};
