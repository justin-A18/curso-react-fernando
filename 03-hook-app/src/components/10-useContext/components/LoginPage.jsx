import { useContext } from 'react';
import { UserContext } from '../context/UserContext';

export const LoginPage = () => {
	const { user, loginUser } = useContext(UserContext);

	const newUser = {
		id: 123,
		name: 'Justin Vargas',
		email: 'justin.hv@gmail.com',
	};

	return (
		<>
			<h1>LoginPage</h1>
			<hr />

			<pre aria-label='code'>{JSON.stringify(user)}</pre>
			<button onClick={() => loginUser(newUser)}>Set user</button>
		</>
	);
};
