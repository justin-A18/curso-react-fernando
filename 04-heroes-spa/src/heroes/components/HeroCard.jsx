import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const CharactersByHero = ({ alter_ego, characters }) => {
	return alter_ego === characters ? (
		<></>
	) : (
		<p className='card-text'>{characters}</p>
	);
};

export const HeroCard = ({
	id,
	superhero,
	alter_ego,
	first_appearance,
	characters,
}) => {
	const imagesURL = `/heroes/${id}.jpg`;

	return (
		<article
			className='col animate__animated animate__fadeIn'
			id={id}>
			<div className='card'>
				<div className='row no-getters'>
					<div className='col-4'>
						<img
							src={imagesURL}
							className='card-img'
							alt={superhero}
						/>
					</div>

					<div className='col-8'>
						<div className='card-body'>
							<h5 className='card-title'>{superhero}</h5>
							<p className='card-text'>{alter_ego}</p>

							<CharactersByHero
								characters={characters}
								alter_ego={alter_ego}
							/>

							<p className='card-text'>
								<small className='text-muted'>{first_appearance}</small>
							</p>

							<Link to={`/hero/${id}`}>Más información</Link>
						</div>
					</div>
				</div>
			</div>
		</article>
	);
};

CharactersByHero.propTypes = {
	alter_ego: PropTypes.string,
	characters: PropTypes.string,
};

HeroCard.propTypes = {
	id: PropTypes.string,
	superhero: PropTypes.string,
	publisher: PropTypes.string,
	alter_ego: PropTypes.string,
	first_appearance: PropTypes.string,
	characters: PropTypes.string,
};
