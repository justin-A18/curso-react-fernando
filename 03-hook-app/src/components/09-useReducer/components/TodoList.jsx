import PropTypes from 'prop-types';
import { TodoItem } from './TodoItem';

export const TodoList = ({ todos = [], onDeleteTodo,onToggleTodo }) => {
	return (
		<ul className='list-group d-flex flex-column gap-4'>
			{todos.map((todo) => (
				<TodoItem
					key={todo.id}
					todo={todo}
					onDeleteTodo={onDeleteTodo}
					onToggleTodo={onToggleTodo}
				/>
			))}
		</ul>
	);
};

TodoList.propTypes = {
	todos: PropTypes.array.isRequired,
	onDeleteTodo: PropTypes.func.isRequired,
	onToggleTodo: PropTypes.func.isRequired
};
