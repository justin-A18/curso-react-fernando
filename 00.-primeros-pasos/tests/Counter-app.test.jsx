import { fireEvent, render,screen } from '@testing-library/react';
import { describe, test, expect } from '@jest/globals';
import { CounterApp } from '../src/components/Counter-app';

describe('Pruebas en Counter App', () => {
	const value = 100;

	test('Esto debe hacer un match con el snapshot', () => {
		const { container } = render(<CounterApp value={value} />);
		expect(container).toMatchSnapshot();
	});

	test('Debe mostrar el Valor inicial de 100', () => {
		render(<CounterApp value={value} />);
		//Screen representacion del componente que se esta reenderizando
		expect(screen.getByText(100)).toBeTruthy();
		//expect(screen.getByRole('heading',{level: 2}).innerHTML).toContain('100');
	});

	test('Debe de incrementar con el boton incremento', () => {
		//con FireEvent Le decimos que evento queremos disparar

		render(<CounterApp value={value} />);
		fireEvent.click(screen.getByText('incremento'));
		expect(screen.getByText('101')).toBeTruthy();
	});

	test('Debe de Decrementar con el boton decremento', () => {
		render(<CounterApp value={value} />);
		fireEvent.click(screen.getByText('decremento'));
		expect(screen.getByText('99')).toBeTruthy();
	});


	test('Debe de Resetear con el boton resetear', () => {
		render(<CounterApp value={value} />);
		fireEvent.click(screen.getByText('resetear'));
		expect(screen.getByText('100')).toBeTruthy();
	});
});
