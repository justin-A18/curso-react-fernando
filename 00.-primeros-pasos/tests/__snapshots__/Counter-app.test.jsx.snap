// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Counter App Esto debe hacer un match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
    decremento
  </button>
  <button>
    resetear
  </button>
  <button>
    incremento
  </button>
</div>
`;
